WBooks/Detail View/DetailViewModel.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  DetailViewModel.swift
+//  WBooks
+//
+//  Created by Gaston Maspero on 30/05/2019.
+//  Copyright © 2019 Wolox. All rights reserved.
+//
+
+import Foundation
+
+enum Availability {
+    case available
+    case notAvailable
+    case inHands
+}
+
+class DetailViewModel {
+    
+    var bookViewModel: BookViewModel
+    
+    var rents: [Rent] = []
+    
+    var status: Availability?
+    
+    init(bookViewModel: BookViewModel) {
+        self.bookViewModel = bookViewModel
+    }
+    
+    func loadRents(onSuccess: @escaping ([Rent]) -> Void) {
+        RentsRepository.fetchRents(onSuccess: onSuccess, onError: onError)
+    }
+    
+    func onLoadRentsSuccess(rents: [Rent]) {
+        self.rents = rents
+        if bookViewModel.status == "Available" || bookViewModel.status == "available" {
+            self.status = .available
+        } else if isBookOnHands() {
+            self.status = .inHands
+        } else {
+            self.status = .notAvailable
+        }
+    }
+    
+    func onError(error: Error) {
+        return
+    }
+    
+    func rentBook(onSuccess: @escaping () -> Void) {
+        let today = Date().toString()
+        let tomorrow = Date().adding(days: 1).toString()
+        RentsRepository.rentBook(onSuccess: onSuccess, bookID: bookViewModel.id, from: today, to: tomorrow)
+    }
+    
+    func onBookRentSuccess() {
+        bookViewModel.status = "rented"
+    }
+    
+    private func isBookOnHands() -> Bool {
+        for rent in rents where bookViewModel.id == rent.book.id {
```

Comment:
Swift helps you with this: use  `.filter(where:)`

Reply:
But why would I need the filtered returned array? I just wanted a linear search here.

Reply:
filter can help you with that. It is not required but is syntactic sugar that helps us to understand what you are doing. with filter is as simple as `return rents.filter(where: { $0.book.id == self.bookViewModel.id }). != nil

Reply:
Ok, I understand. But If the for loop would end as soon as that condition is met. The filter would process the whole array. If there were a lot of rents, wouldn't it be slower?
I can see the syntactic sugar though.

Reply:
Yes, you are right. Maybe `first(where:)` would be a better option than the filter

Reply:
ok!